buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiTask.java

```diff
@@ -117,21 +122,21 @@ public FileTree getInputDir() {
         FileTree xpackFileTree = null;
         if (includeXpack.get().isEmpty() == false) {
             xpackPatternSet.setIncludes(includeXpack.get().stream().map(prefix -> prefix + "*/**").collect(Collectors.toList()));
-            xpackFileTree = xpackConfig.getAsFileTree().matching(xpackPatternSet);
+            xpackFileTree = xpackConfigToFileTree.apply(xpackConfig).matching(xpackPatternSet);
         }
         boolean projectHasYamlRestTests = skipHasRestTestCheck || projectHasYamlRestTests();
         if (includeCore.get().isEmpty() == false || projectHasYamlRestTests) {
             if (BuildParams.isInternal()) {
                 corePatternSet.setIncludes(includeCore.get().stream().map(prefix -> prefix + "*/**").collect(Collectors.toList()));
-                coreFileTree = coreConfig.getAsFileTree().matching(corePatternSet); // directory on disk
+                coreFileTree = coreConfigToFileTree.apply(coreConfig).matching(corePatternSet); // directory on disk
             } else {
                 coreFileTree = coreConfig.getAsFileTree(); // jar file
             }
         }
 
         ConfigurableFileCollection fileCollection = additionalConfig == null
             ? getProject().files(coreFileTree, xpackFileTree)
```

Comment:
Not strictly related but since we're here: can we change this task to use the `ProjectLayout` Service here instead of referencing the project object. This helps us moving (slowly) towards supporting the gradle configuration cache (https://docs.gradle.org/current/userguide/configuration_cache.html) You can inject the service e.g. via constructor and annotating that constructor with @Inject